src/API/Management/Users.php

```diff
@@ -1,88 +1,265 @@
+        }
+
+        if ('sms' === $data['connection'] && empty($data['phone_number'])) {
+            // "phone_number" field is required for an sms connection.
+            throw new \Exception('Missing required "phone_number" field for sms connection.');
+        } elseif ('sms' !== $data['connection']) {
+            // "email" field is required for email and DB connections.
+            if (empty($data['email'])) {
+                throw new \Exception('Missing required "email" field.');
+            }
+
+            // Passwords are required for DB connections.
+            if ('email' !== $data['connection'] && empty($data['password'])) {
+                throw new \Exception('Missing required "password" field for "' . $data['connection'] . '" connection.');
+            }
+        }
 
-        return $this->apiClient->post()
-            ->users()
-            ->withHeader(new ContentType('application/json'))
+        return $this->apiClient->method('post')
+            ->addPath('users')
             ->withBody(json_encode($data))
             ->call();
     }
 
-    public function getAll($params = array()) {
-        return $this->search($params);
-    }
-
-    public function search($params = array()) {
+    /**
+     * Search all Users.
+     * Requires scopes: read:users, read:user_idp_tokens.
+     *
+     * @param array $params - Search parameters to send:
+     *      - "fields", "include_fields", "page", and "per_page" keys here will override the explicit parameters.
+     *      - Queries using "search_engine" set to "v2" should be migrated to v3; see search v3 @link below.
+     * @param null|string|array $fields - Fields to include or exclude from the result, empty to retrieve all fields:
+     *      - Including only the fields required can speed up API calls significantly.
+     *      - Arrays will be converted to comma-separated strings
+     * @param null|boolean $include_fields - True to include $fields, false to exclude $fields.
+     * @param null|integer $page - Page number to get, zero-based.
+     * @param null|integer $per_page - Number of results to get, null to return the default number.
+     *
+     * @return mixed
+     *
+     * @throws \Exception
+     *
+     * @link https://auth0.com/docs/users/search/v3
+     * @link https://auth0.com/docs/api/management/v2#!/Users/get_users
+     */
+    public function getAll($params = [], $fields = null, $include_fields = null, $page = null, $per_page = null)
+    {
+        $params = is_array($params) ? $params : [];
 
-        $client = $this->apiClient->get()
-            ->users();
+        // Fields to include/exclude.
+        if (!isset($params['fields']) && null !== $fields) {
+            $params['fields'] = $fields;
+        }
+        if (isset($params['fields'])) {
+            if (is_array($params['fields'])) {
+                $params['fields'] = implode(',', $params['fields']);
+            }
+            if (null !== $include_fields) {
```

Comment:
here you're not checking that `include_fields` is first set on `params`, so you're overriding it with the explicit parameter.